csbuild/tools/java_compilers/oracle_java_compiler.py

```diff
@@ -45,11 +44,7 @@ def __init__(self, projectSettings):
 		JavaCompilerBase.__init__(self, projectSettings)
 
 		self._javaCompilerPath = os.path.join(self._javaBinPath, "javac{}".format(".exe" if platform.system() == "Windows" else ""))
-
-		try:
-			subprocess.call([self._javaCompilerPath], stdout = subprocess.PIPE, stderr = subprocess.PIPE)
-		except:
-			raise IOError("Oracle Java compiler not found at path: {}".format(self._javaCompilerPath))
+		assert os.access(self._javaCompilerPath, os.X_OK), "Oracle Java compiler not found at path: {}".format(self._javaCompilerPath)
```

Comment:
I'm not sure whether or not X_OK works properly on Windows. On linux, it's literally just checking if the executable permission is set - doesn't actually confirm anything about it being able to execute SUCCESSFULLY. You can set that bit on any file. Maybe that's fine, just wanted to make sure to point it out.

Reply:
Also as a sidebar - on linux, any directory you can enter will also return true for X_OK. Not sure if you care about the edge case where the path exists but is a directory. Seems unlikely to happen.

Reply:
I think Windows considers all files to be executable as far as that bit is concerned. I tested on Windows and it works correctly.  In this case, I think it's sufficient to just check for the existence of the file and the ability to execute it.  If it can't actually be run, that will be caught when attempting to run it.

Reply:
Works for me.